simpledotorg/simple-server
# 2795 incorrect registrations count on overview

**Story card:** [ch2795](https://app.clubhouse.io/simpledotorg/story/2795/registration-counts-are-inconsistent-across-details-and-overview-tabs)

## Because

The registrations count in the reports overview can be wrong. 

## This addresses

Currently, when there are no registrations for the current month, the last month's registrations data can appear in this month's place. This is because of the way that registrations and assigned patients are queried and grouped by month and then processed through the Region::Result object. 

## Test instructions

The bug only appears when there are no registrations in the current month, so it may be difficult to repro if you have re-seeded your db recently. 
- Turn on the  report_with_exclusions flipper flag.
- Go to the reports details page and look in "Patient registrations and follow-ups" monthly registered patients for the current month. If that's a dash, we would expect this bug to occur. 
- Go to the reports overview page and look at "Mar-2021 district performance". The "Patients registered in Mar-2021" column should be 0 if you saw a dash in the details page.
